field/TinyLang/Field/Raw/Core.hs

```diff
@@ -24,16 +26,24 @@ newtype Var = Var { unVar :: Identifier }
     deriving (Eq, Generic)
     deriving (Show) via (Quiet Var)
 
+{-| @Statements v f$, @Statement v f@, and @Expr v f@ are parameterised
+  by the type of variable @v@.
+-}
+newtype Statements v f = Statements { unStatements :: [Statement v f] }
```

Comment:
I'd call this `Program` and not use the wrapper in `For`. So that we can have `parseProgram`, `checkProgram`, etc.